EventGenerator/src/CaloCalibGun_module.cc

```diff
@@ -0,0 +1,241 @@
+  }
+
+  //================================================================
+  void CaloCalibGun::produce(art::Event& event) {
+    std::unique_ptr<GenParticleCollection> output(new GenParticleCollection);
+    PrimaryParticle primaryParticles;
+    MCTrajectoryCollection mctc;
+
+    const DiskCalorimeter *_cal = GeomHandle<DiskCalorimeter>().get();
+    //_nPipes          = _cal->caloInfo().getInt("nPipes");
+    _pipeRadius      = _cal->caloInfo().getDouble("pipeRadius");
+    _pipeTorRadius   = _cal->caloInfo().getVDouble("pipeTorRadius");
+    _randomRad       = _cal->caloInfo().getVDouble("pipeTorRadius");
+    _zPipeCenter     = _cal->disk(_nDisk).geomInfo().origin()-CLHEP::Hep3Vector(0,0,_cal->disk(_nDisk).geomInfo().size().z()/2.0-_pipeRadius);//disk =1
+    // we normalize to the volume of the pipe (proportional to 2*pi*R if they have all the same radius) to draw a random number from which to generate the photons TODO - is this used?
+    double sumR(0);
+    std::for_each(_randomRad.begin(), _randomRad.end(), [&](double& d) {sumR+=d; d = sumR; });
+    std::for_each(_randomRad.begin(), _randomRad.end(), [&](double& d) {d /= sumR;});
+ 
+    //Define the parameters of the pipes:
+    
+    // angle of large torus in degrees
+    std::vector<double> phi_lbd = _cal->caloInfo().getVDouble("largeTorPhi");
+    // angle of small torus in degrees
+    std::vector<double> phi_sbd = _cal->caloInfo().getVDouble("smallTorPhi");
+    // angle of the end point
+    std::vector<double> phi_end = _cal->caloInfo().getVDouble("straightEndPhi");
+    // center position y of the small torus
+    std::vector<double> ysmall = _cal->caloInfo().getVDouble("yposition");
+    // radius of small torus
+    double radSmTor = _cal->caloInfo().getDouble("radSmTor");
+    // first center position x of the small torus
+    double xsmall = _cal->caloInfo().getDouble("radSmTor");
+    // distance of the small torus center
+    double xdistance = _cal->caloInfo().getDouble("xdistance");
+    // inner radius of the manifold
+    double rInnerManifold = _cal->caloInfo().getDouble("rInnerManifold");
+    vector<float> sign{-1.0, 1.0};
+
+    for (unsigned int ig = 0; ig < _mean; ++ig) 
+    {
+      double xpipe, ypipe, zpipe;
+      //Pick position - find either 0,1 - these are indices of the sign list (so 0=-1, 1=+1)
+      int xsn = round(_randFlat.fire()); 
+      int ysn = round(_randFlat.fire());
+
+      // pick a random theta, between 0 and 2*pi:
+      double theta = _randFlat.fire() * 2.0 * CLHEP::pi;
+      // pick a random point on the radius
+      double pipeR = _pipeRadius * sqrt(_randFlat.fire());
+      // find the z position based on above:
+      zpipe = pipeR*sin(theta);
+
+      double rtest = _randFlat.fire();
+      // select an index from list of pipes:
+      int idx = int( std::lower_bound(_randomRad.begin(), _randomRad.end(), rtest) - _randomRad.begin());
```

Comment:
As I read the code, this uses the radius of the large torus as a proxy for the volume of the: large torus + small torus + straight section.  Is that right?